File: Models/Expense.cs
using System;
using System.Runtime.Serialization;
using ServiceStack.DataAnnotations;


namespace CSVisualizer.Models{
    [DataContract]
    public class ExpenseTemp{
        [DataMember]
        [Index(Unique = true)]
        public string date{get;set;}
        [DataMember]
        public string category{get;set;}
        [DataMember(Name="employee name")]
        public string employee_name{get;set;}

        [DataMember(Name="employee address")]
        public string employee_address{get;set;}
        [DataMember(Name="expense description")]
        public string expense_description{get;set;}
        [DataMember(Name="pre-tax amount")]
        public string pre_tax_amount {get;set;}
        [DataMember(Name="tax name")]
        public string tax_name {get;set;}
        [DataMember(Name="tax amount")]
        public string tax_amount{get;set;}


    }

    public class MonthlyExpense{
        public string Month{get;set;}
        public decimal Expense{get;set;}

        public MonthlyExpense(string _month,decimal _expense){
            Month=_month;
            Expense=_expense;
        }


    }
    public class Employee {
        [AutoIncrement]
        [PrimaryKey]
        public int EmployeeID {get;set;}
        public string EmployeeName {get;set;}
        public string EmployeeAddress{get;set;}
 }

    public class ExpenseType {
        [AutoIncrement]
         [PrimaryKey]
        public int ExpenseTypeID {get;set;}
       
        public string ExpenseCategory{get;set;}

     
    }
    public class TaxType {
        [AutoIncrement]
        [PrimaryKey]
        public int TaxTypeID{get;set;}
        public string TaxName{get;set;}
    }

    public class EmployeeExpense{
        [AutoIncrement]
        [PrimaryKey]
        public int EmployeeExpenseID{get;set;}
        [ForeignKey(typeof(Employee))]
        public int EmployeeID{get;set;}
        [ForeignKey(typeof(ExpenseType))]
        public int ExpenseTypeID{get;set;}

        public DateTime ExpenseDate{get;set;}
        [ForeignKey(typeof(TaxType))]
        public int TaxID{get;set;}

        public decimal PreTaxAmnt{get;set;}

        public decimal TaxAmnt {get;set;}

     

    }


}

File: app/models/Expense.py
# -*- encoding: utf-8 -*-
from app import db
from sqlalchemy.ext.hybrid import hybrid_property


class Expense(db.Model):
    # id
    id = db.Column(db.Integer, primary_key=True, autoincrement=True)

    # date
    date = db.Column(db.Date)

    # category
    category_id = db.Column(db.Integer, db.ForeignKey('category.id'))
    category = db.relationship(
        'Category',
        backref=db.backref('expenses', lazy='dynamic')
    )

    # employee
    employee_id = db.Column(db.Integer, db.ForeignKey('employee.id'))
    employee = db.relationship(
        'Employee',
        backref=db.backref('expenses', lazy='dynamic')
    )

    # tax
    tax_id = db.Column(db.Integer, db.ForeignKey('tax.id'))
    tax = db.relationship('Tax')

    # expense description
    description = db.Column(db.String(255))

    # pre-tax amount
    pre_tax_amount = db.Column(db.Float)

    def __init__(self, date, category, employee, tax, description,
                 pre_tax_amount):
        self.date = date
        self.category = category
        self.employee = employee
        self.tax = tax
        self.description = description
        self.pre_tax_amount = pre_tax_amount

    def __repr__(self):
        return '<Expense %r>' % self.description

    @hybrid_property
    def post_tax_amount(self):
        return self.pre_tax_amount * (1.0 + self.tax.amount)
